### PR TITLE
Rename load-balancer by proxy

### PR DESCRIPTION
**Pull request by dbengsch**

### CODE REVIEW

File: test/core/loadBalancer.test.js [hunk 1]
```diff
@@ -3,12 +3,12 @@ var
   should = require('should'),
   rewire = require('rewire'),
   path = require('path'),
-  LoadBalancer = rewire('../../lib/core/LoadBalancer'),
+  LoadBalancer = rewire('../../lib/core/KuzzleProxy'),
```
Comment (ballinette) [inline test/core/loadBalancer.test.js]: a renaming missing:
`LoadBalancer = rewire('../../lib/core/KuzzleProxy')`
=> `KuzzleProxy = rewire('../../lib/core/KuzzleProxy')`

Comment (ballinette) [inline test/core/loadBalancer.test.js]: (and also rename the file into kuzzleProxy.test.js)

Comment (dbengsch) [inline test/core/loadBalancer.test.js]: Done

**Comment by ballinette** [comment]
found 2 missing instances of term "load balancer" within documentation files and comments.
I've directly fixed them, so now, +1 ;)
(as soon as #6 is merged, of course)

**Comment by ballinette** [comment]
Oh, just one remark/opened question: 
We still have references to "kuzzleio/lb-base" and "kuzzleio/lb-test" docker images within Dockerfile and .travis.yml.
Should we keep these images naming, or also rename them ?

**Comment by stafyniaksacha** [comment]
LGTM

**Comment by dbengsch** [comment]
@ballinette : The issue about "lb" on the kuzzle side is fixed by this PR : https://github.com/kuzzleio/kuzzle/pull/300

**Comment by ballinette** [comment]
What I meant is:
We still have reference to "lb" docker images, not on the kuzzle side, but into current repo:
https://github.com/kuzzleio/kuzzle-proxy/blob/lb-to-proxy/.travis.yml
https://github.com/kuzzleio/kuzzle-proxy/blob/lb-to-proxy/Dockerfile

So I guess these 2 files need also to be fixed ?

**Comment by dbengsch** [comment]
@ballinette : Oh yes, I thought it was already done, my bad. Now it is.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
-# kuzzle-load-balancer
+# Kuzzle Proxy
 
-Node.js load-balancer and multiplexer
+Node.js proxy and multiplexer
```
Comment (scottinet) [inline README.md]: Nitpicking: shouldn't this be "Kuzzle proxy and multiplexer"?

**Comment by scottinet** [comment]
+1

**Pull request merged**